corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -223,20 +225,35 @@ hqDefine("geospatial/js/geospatial_map", [
                 map = loadMapBox();
             }
 
-            $exportButton.click(function(e) {
+            $exportButton.click(function () {
                 if (map) {
                     exportGeoJson(map.getMapboxDrawInstance());
                 }
             });
 
             if ($data.length && map) {
-                var caseData = $data.data("context");
+                var contextData = $data.data("context");
                 map.clearMap();
-                cases = caseData.cases
+                cases = contextData.cases;
                 map.addCaseMarkersToMap();
+
+                if (contextData.invalid_geo_cases_report_link) {
+                    var missingCasesLink = contextData.invalid_geo_cases_report_link;
+                    var missingCasesLinkTag = "<a href=" + missingCasesLink + ">" + gettext("View here") + "</a>";
+                    var message = gettext("There are case(s) missing geolocation data.");
+
+                    alert_user.alert_user(message + " " + missingCasesLinkTag, "warning");
+
+                    var $bannerAlert = $("#message-alerts");
+                    if ($bannerAlert.children().length > 1) {
+                        // Remove the initial banner, since it contains the old link
+                        $bannerAlert.children()[0].remove();
```

Comment:
The `if` clause checks that there is more than one child. Are you assuming that there are only two children? Are you sure?

Reply:
@kaapstorm 
What happens is, when `alert_user` is called a child element is added to the banner container div element. The moment there's more than one child element this conditional block will get executed, hence I think the assumption of only two child elements is fairly safe. 

The only risk is having more than one banner showing on the screen in case my assumption fails.